cmd/helmExecute.go

```diff
@@ -120,3 +129,42 @@ func runHelmExecuteDefault(config helmExecuteOptions, helmExecutor kubernetes.He
 
 	return nil
 }
+
+//getAndRenderImageInfo gets data from commonPipelineEnvironment
+func getAndRenderImageInfo(config helmExecuteOptions, rootPath string, utils kubernetes.DeployUtils) error {
+	cpe := piperenv.CPEMap{}
+	err := cpe.LoadFromDisk(path.Join(rootPath, "commonPipelineEnvironment"))
+	if err != nil {
+		return fmt.Errorf("failed to load values from commonPipelineEnvironment: %v", err)
+	}
+
+	valuesFiles := []string{fmt.Sprintf("%s/%s", config.ChartPath, "values.yaml")}
+	valuesFiles = append(valuesFiles, config.HelmValues...)
+
+	params := struct {
+		CPE map[string]interface{}
+	}{
+		CPE: cpe,
+	}
```

Comment:
I wonder if it is a good idea to freely pass the CPE as a whole into the Helm `values.yaml` file that will be packaged, uploaded to the staging service and later promoted to the internal or internet facing common repository.

Isn't it to easy for Helm chart authors to accidentially leak confidential data that should not leave the CPE?

Reply:
I guess this is okay, since the CPE does not really contain confidential data.

Nonetheless, it probably makes sense to add some template functions to better be able to access values.

Example: today it will be rather difficult to get the correct sha for an image if there is more than one available in the CPE.
Having a template function to get the sha for an image name could be very beneficial, I guess.
There might be more cases where a dedicated template function could make sense.

Reply:
I like the idea of having cpe lookup functions.

Reply:
I created a PR with some re-use capability which we could consider to use here as well ...

Reply:
Added 'temporary' template functions to have better access to some CPE values - [14a1a9f](https://github.com/SAP/jenkins-library/pull/3861/commits/14a1a9fcfeb3b6d2ace4d0e72f303ee495fd84d0). 
Will use [#3872](https://github.com/SAP/jenkins-library/pull/3872) functionality once it is merged

Reply:
Added usage of [#3872](https://github.com/SAP/jenkins-library/pull/3872) functionality - [172e792](https://github.com/SAP/jenkins-library/pull/3861/commits/172e792cafc59de72cffa39cef46d2055c399490)